internal/collector/instance.go

```diff
@@ -99,6 +105,32 @@ func AddToPod(
 		VolumeMounts:    append(volumeMounts, configVolumeMount),
 	}
 
+	// If a retentionPeriod is set and this is a pod that uses logrotate for
+	// log rotation, add config volume and mount for logrotate config
+	if includeLogrotate && spec != nil && spec.Logs != nil && spec.Logs.RetentionPeriod != nil {
```

Comment:
Do we require `RetentionPeriod` or are we allowing that to be blank and we'll default to a day?

Reply:
We allow it to be blank and default to a day.

Reply:
I'll remove the check for `spec` etc here